nerfstudio/configs/annotated_types.py

```diff
@@ -0,0 +1,159 @@
+# Copyright 2022 the Regents of the University of California, Nerfstudio Team and contributors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import dataclasses
+from collections import OrderedDict
+from typing import TYPE_CHECKING
+
+import tyro
+
+from nerfstudio.configs.dataparser_configs import dataparser_configs
+from nerfstudio.configs.external_methods import get_external_methods
+from nerfstudio.configs.method_configs import descriptions as method_descriptions
+from nerfstudio.configs.method_configs import method_configs
+from nerfstudio.data.dataparsers.base_dataparser import DataParserConfig
+from nerfstudio.engine.trainer import TrainerConfig
+from nerfstudio.plugins.registry import discover_dataparsers, discover_methods
+
+
+def merge_methods(methods, method_descriptions, new_methods, new_descriptions, overwrite=True):
+    """Merge new methods and descriptions into existing methods and descriptions.
+    Args:
+        methods: Existing methods.
+        method_descriptions: Existing descriptions.
+        new_methods: New methods to merge in.
+        new_descriptions: New descriptions to merge in.
+    Returns:
+        Merged methods and descriptions.
+    """
+    methods = OrderedDict(**methods)
+    method_descriptions = OrderedDict(**method_descriptions)
```

Comment:
Would `method_descriptions.copy()` work here? (since we require Python >=3.8 `OrderedDict` also seems unnecessary, unless it's just to make the intention clear)

Reply:
I agree, no need to keep using OrderedDict since python>=3.8 is enforced.